Code/Mantid/Framework/Algorithms/src/ModeratorTzero.cpp

```diff
@@ -329,6 +500,26 @@ double ModeratorTzero::CalculateT0(const double &tof, const double &L1,
   return t0_next;
 }
 
+/// Calculate emission time for a given detector (L1, t2)
+/// and TOF when Emode==Elastic
+double ModeratorTzero::CalculateT0elastic(const double &tof, const double &L12,
+  double &E1, mu::Parser &parser)
+{
+  double t0_curr, t0_next;
+  t0_curr = m_tolTOF; // current iteration emission time
+  t0_next = 0.0;      // next iteration emission time, initialized to zero
+  size_t iiter(0);    // current iteration number
+  // iterate until convergence in t0 reached
+  while (std::fabs(t0_curr - t0_next) >= m_tolTOF && iiter < m_niter)
+  {
+    t0_curr = t0_next;
+    double v1 = L12 / tof; // v1 = v2 = v since emode is elastic
```

Comment:
This doesn't appear to update v1 every iteration. I think this needs to be `v1 = L12 / (tof - t0_curr)`.